### PR TITLE
Add attachToCompileJava flag for gradle 6.x compat

### PR DESCRIPTION
**Pull request by Sineaggi**
Prevents `Removing a task dependency from a task instance is not supported.` in gradle 6.0-rc-1.

### CODE REVIEW

**Comment by etiennestuder** [comment]
Please add a test case if you want this PR to be considered.

File: src/main/groovy/nu/studer/gradle/jooq/JooqExtension.groovy [hunk 1]
```diff
@@ -31,6 +31,7 @@ class JooqExtension {
     final Closure whenConfigAdded
     final String path
     final Map<String, JooqConfiguration> configs
+    Boolean attachToCompileJava = true
```
Comment (etiennestuder) [inline src/main/groovy/nu/studer/gradle/jooq/JooqExtension.groovy]: Why the Wrapper type?

**Comment by Leichtwind** [comment]
@etiennestuder I am not sure that @Sineaggi is currently active and will apply the required changes. Don't you mind if I open another PR?

File: src/main/groovy/nu/studer/gradle/jooq/JooqExtension.groovy (hunk 1 above)
Comment (Sineaggi) [reply src/main/groovy/nu/studer/gradle/jooq/JooqExtension.groovy]: I've been writing too much kotlin.

**Comment by Leichtwind** [comment]
@Sineaggi test case [`can remove auto-wired task dependency`](https://github.com/Sineaggi/gradle-jooq-plugin/blob/2d2d525f594612f8dd08fd9cb3414a6faced3ea2/src/test/groovy/nu/studer/gradle/jooq/JooqFuncTest.groovy#L163) still uses direct removal of task dependency. Please update the test case as well to use `attachToCompileJava` instead of removing the task dependency
Or you can add me to your fork as contributor so I could do it myself

**Comment by Leichtwind** [comment]
@etiennestuder I have updated the test case. Can you please make a review?

**Comment by etiennestuder** [comment]
I understand the code, but what is the underlying use case for removing the task dependency?

**Comment by Leichtwind** [comment]
@etiennestuder Migrations in the project are run using Liquibase on application start and after the jOOQ schema source generation task is completed. That makes application quite inconvenient to run if there are new migrations. So it is easier to run jOOQ schema source generation task when it is required

**Comment by etiennestuder** [comment]
Do I understand this right: you want to run the source code generation explicitly and then you check in the generated sources into your code repository?

**Comment by Leichtwind** [comment]
Yes, you do understand this right

**Comment by etiennestuder** [comment]
OK. I'll think about it over the coming days. I believe there is another way to solve this.

**Comment by adamkobor** [comment]
> @etiennestuder Migrations in the project are run using Liquibase on application start and after the jOOQ schema source generation task is completed. That makes application quite inconvenient to run if there are new migrations. So it is easier to run jOOQ schema source generation task manually when it is required

Our use case is quite similar to that, and since we would like to upgrade our builds to Gradle 6, it's definitely a showstopper :(

**Comment by etiennestuder** [comment]
Ideally, there would be a flag per configuration instead of a global flag. But, it is easy enough to turn off auto-wiring globally and manually add a task dependency for a specific configuration.

The flag will be available in 4.1.

**Comment by etiennestuder** [comment]
Thanks for the PR.

**Pull request merged**